geopandas/io/file.py

```diff
@@ -450,6 +485,37 @@ def _read_file_pyogrio(path_or_bytes, bbox=None, mask=None, rows=None, **kwargs)
     if kwargs.pop("ignore_geometry", False):
         kwargs["read_geometry"] = False
 
+    # translate `ignore_fields`/`include_fields` keyword for back compat with fiona
+    if "ignore_fields" in kwargs and "include_fields" in kwargs:
+        raise ValueError("Cannot specify both 'ignore_fields' and 'include_fields'")
+    elif "ignore_fields" in kwargs:
+        if kwargs.get("columns", None) is not None:
+            raise ValueError(
+                "Cannot specify both 'columns' and 'ignore_fields' keywords"
+            )
+        warnings.warn(
+            "It is recommended to use the 'columns' keyword to select the columns "
```

Comment:
Changed it to a deprecation warning (at least already on the pyogrio code path, not yet for fiona because there it also depends on the fiona version).

We could relatively simply keep those keywords, but it also just adds to the API bloat (if we keep them (beyond temporary back compat), we should also document them properly etc)

Reply:
I think we seem to be on the same page now, let's warn with the intent to remove and save ourselves the headache of maintaining two apis.